src/v/cluster_link/task.cc

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster_link/task.h"
+
+#include "cluster_link/errc.h"
+#include "cluster_link/logger.h"
+#include "ssx/future-util.h"
+
+namespace cluster_link {
+
+task::task(ss::lowres_clock::duration run_interval, ss::sstring name)
+  : _run_interval(run_interval)
+  , _name(std::move(name))
+  , _logger(cllog, _name) {}
+
+ss::future<result<void>> task::start() {
+    vlog(logger().trace, "start called");
+    auto prev_state = BOOST_OUTCOME_CO_TRYX(change_state(
+      model::task_state::active, ssx::sformat("{} has started", name())));
+    /// Start can be be used to unpause a task that was previously paused
+    if (prev_state == model::task_state::not_running) {
+        _as.emplace();
+        vlog(logger().debug, "Initial start of task - beginning fiber");
+        ssx::background = ss::with_gate(_gate, [this] { return run_task(); });
+    }
+
+    co_return outcome::success();
+}
+
+ss::future<result<void>> task::stop() {
+    vlog(logger().trace, "stop called");
+    auto res = change_state(
+      model::task_state::not_running, ssx::sformat("{} has stopped", name()));
+    vassert(res.has_value(), "Failed to change state to not_running");
+    if (_as.has_value()) {
+        _as->request_abort();
+    }
+    co_await _gate.close();
+    _as = std::nullopt;
```

Comment:
based on our experience with net::transport, it can be really tricky to work with processes that can be restarted (e.g. resetting an abort source), not impossible of course, just foot guns around. one strategy for simplifying it is to factor out the bits into a smaller sub-process/fiber that only has start/stop semantics, and then at a higher level (e.g. at the task level) model pause/resume as stop + start of a new instance (where the new instance is configured as-if it were being resumed).